backend/danswer/connectors/confluence/connector.py

```diff
@@ -286,17 +442,42 @@ def _fetch(start_ind: int, batch_size: int) -> Collection[dict[str, Any]]:
 
                 return view_pages
 
+        def _fetch_page(batch_size: int) -> list[dict[str, Any]]:
+            indexer = RecursiveIndexer(
```

Comment:
this will cause us to re-fetch all pages every batch, no? We should probably just do this once for a given run

Reply:
Yep! Is/was only called once, but updated the naming to make this clearer

Reply:
Ah, I think for clarity / consistency it would be great if `_fetch_pages` always returned a list of pages of length `self.batch_size`. I see we don't do that here.

This also has some additional benefits, such as only fetching comments as needed, and then removing them from memory once we're done with the batch. LMK if that makes sense!